src/lib.nr

```diff
@@ -200,13 +223,132 @@ fn test_base64_decode_elements() {
+    ];
+    let result:[u8; 610] = base64_decode(input);
+    let expected = "The quick brown fox jumps over the lazy dog, while 42 ravens perch atop a rusty mailbox. Zany quilters fabricate 9 cozy blankets, as 3 jovial wizards expertly mix 5 potent elixirs. Bright neon signs flash \"OPEN 24/7\" in the misty night air, illuminating 8 vintage cars parked along Main Street. A gentle breeze carries the aroma of fresh coffee and warm cinnamon rolls from Joe's Diner, enticing 6 sleepy truckers to stop for a late-night snack. Meanwhile, 11 mischievous kittens playfully chase a ball of yarn across Mrs. Johnson's porch, their antics observed by 2 wise old owls perched on a nearby oak tree.";
+
+    assert(result == expected.as_bytes());
+}
+
+// TODO: support padding
+// this test should fail, because padding is not supported
```

Comment:
Padding for decode should work fine if we return 0 for 61 index in the `Base64DecodeBE` right?

It is a problem in encoding though, as we are not adding the padding chars to the encoded results currently. For this we can do something [like this](https://github.com/noir-lang/noir_base64/pull/11/files#diff-7d9659023f15743be35b9fdc7638990bb804588e410a7f778802dd2be3c87707R183-R193) (will be closing that PR in favour of yours btw)

What do you think? If that make sense we can do padding fix as a follow up PR to this.

Reply:
What I meant by this comment is that it is actually correct for the padding character to cause a failure until it is implemented (as you've done in your PR). Even once padding is implemented, if 61 is encountered anywhere before the end, this would be an invalid character. I agree that implementing padding would be a good follow-up PR. Imo, the ideal case is to make padding the default but have an option to omit padding, since in many cases it's not required and will be more efficient without. I did an implementation of this in my own repo [here](https://github.com/grjte/noir_base64_standard/blob/main/src/encoder.nr), where I also added support for the base64url alphabet (which I need). In the base64url alphabet, it will also be correct for 61 to always cause a failure, since the padding character is different.